pool.go

```diff
@@ -32,6 +33,8 @@ type Pool struct {
 	// exit is used to signal the pool to stop running the query loop
 	// and close the jobs channel
 	exit chan bool
+	// afterFunc is a function that gets called after each job.
+	afterFunc *func(*Job)
```

Comment:
@albrow didn't know about that trick. Neat!
Yeah, that's probably not needed then.